contracts/example/tests/test.rs

```diff
@@ -1,69 +1,153 @@
+    its_client: InterchainTokenServiceClient<'a>,
+    app: ExampleClient<'a>,
+}
 
-    client
+fn setup_app<'a>(env: &Env, chain_name: &String) -> TestConfig<'a> {
+    let (signers, gateway_client) = setup_gateway(&env);
+    let gas_service_client = setup_gas_service(&env);
+    let its_client = setup_its(
+        &env,
+        &gateway_client.address,
+        &gas_service_client.address,
+        &chain_name,
+    );
+    let app_address = env.register(
+        Example,
+        (
+            &gateway_client.address,
+            &gas_service_client.address,
+            &its_client.address,
+        ),
+    );
+    let app = ExampleClient::new(env, &app_address);
+
+    TestConfig {
+        signers,
+        gateway_client,
+        gas_service_client,
+        its_client,
+        app,
+    }
+}
+
+fn setup_its<'a>(
+    env: &Env,
+    gateway: &Address,
+    gas_service: &Address,
+    chain_name: &String,
+) -> InterchainTokenServiceClient<'a> {
+    let owner = Address::generate(env);
+    let operator = Address::generate(env);
+    let its_hub_address = String::from_str(env, ITS_HUB_ADDRESS);
+    let native_token_address = Address::generate(env);
+    let interchain_token_wasm_hash = env
+        .deployer()
+        .upload_contract_wasm(INTERCHAIN_TOKEN_WASM_HASH);
+
+    let its_address = env.register(
+        InterchainTokenService,
+        (
+            &owner,
+            &operator,
+            gateway,
+            gas_service,
+            its_hub_address,
+            chain_name.clone(),
+            native_token_address,
+            interchain_token_wasm_hash,
+        ),
+    );
+
+    let its_client = InterchainTokenServiceClient::new(env, &its_address);
+
+    its_client
+}
+
+fn setup_its_token(
+    env: &Env,
+    client: &InterchainTokenServiceClient,
+    sender: &Address,
+    supply: i128,
+) -> BytesN<32> {
+    let salt = BytesN::from_array(env, &[1u8; 32]);
+    let token_metadata = TokenMetadata {
+        name: String::from_str(env, "Test"),
+        symbol: String::from_str(env, "TEST"),
+        decimal: 18,
+    };
+
+    let token_id = client.mock_all_auths().deploy_interchain_token(
+        sender,
+        &salt,
+        &token_metadata,
+        &supply,
+        &None,
+    );
+
+    token_id
 }
 
 #[test]
 fn gmp_example() {
     let env = Env::default();
-    env.mock_all_auths();
 
     let user: Address = Address::generate(&env);
 
     // Setup source Axelar gateway
-    let source_chain = String::from_str(&env, "source");
-    let (_, source_gateway_client) = setup_gateway(&env);
-    let source_gateway_id = source_gateway_client.address;
-    let (source_gas_service_client, _source_gas_collector, source_gas_service_id) =
-        setup_gas_service(&env);
-    let source_app = setup_app(&env, &source_gateway_id, &source_gas_service_id);
+    let source_chain = String::from_str(&env, SOURCE_CHAIN_NAME);
+    let source_test_config = setup_app(&env, &source_chain);
```

Comment:
```suggestion
    let TestConfig { .. } = setup_app(&env, &source_chain);
```
you can destructure it to extract the fields you want with an appropriate name